lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3726,21 +3726,32 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
 // deserialized before loading the named member of this class. This allows the
 // decl members table to be warmed up and enables the correct identification of
 // overrides.
-//
-// FIXME: Very low hanging fruit: Loading everything is extremely wasteful. We
-// should be able to just load the name lazy member loading is asking for.
-static void ensureSuperclassMembersAreLoaded(const ClassDecl *CD) {
+static void loadNamedMemberOfSuperclassIfNeeded(const ClassDecl *CD,
+                                                DeclBaseName name) {
   if (!CD)
     return;
 
   CD = CD->getSuperclassDecl();
-  if (!CD || !CD->hasClangNode())
+  if (!CD)
     return;
-  
-  CD->loadAllMembers();
 
-  for (auto *ED : const_cast<ClassDecl *>(CD)->getExtensions())
-    ED->loadAllMembers();
+  auto &ctx = CD->getASTContext();
```

Comment:
I think this can actually be replaced with a call to `(void) CD->lookupDirect(name)`. I'm going to test and benchmark both versions before checking this in.

Reply:
Yes, it is.